### ISSUE TITLE
Update custom ComponentContext docs for ChildOverlay

### ISSUE THREAD

**Issue by azlekov** [arkivanov/Decompose#232]
Hello, 

Similar to the `childStack` extension functions where custom component can be specified, it would be great to have an example in the docs for the `childOverlay`.

I'm willing to open a PR soon.

Meanwhile:

```kotlin
inline fun <reified C : Parcelable, T : Any> AppComponentContext.appChildOverlay(
    source: OverlayNavigation<C>,
    noinline initialConfiguration: () -> C?,
    key: String = "AppDefaultOverlay",
    persistent: Boolean = false,
    handleBackButton: Boolean = false,
    noinline childFactory: (configuration: C, AppComponentContext) -> T
) = childOverlay(
    source = source,
    configurationClass = C::class,
    key = key,
    initialConfiguration = initialConfiguration,
    persistent = persistent,
    handleBackButton = handleBackButton,
) { configuration, componentContext ->
    childFactory(
        configuration,
        DefaultAppComponentContext(
            componentContext = componentContext,
        )
    )
}
```

**Comment by arkivanov** [comment]
Yeah, I missed this one!

**Comment by arkivanov** [comment]
Feel free to send a PR!

**Comment by azlekov** [comment]
Later today it will be.

**Comment by arkivanov** [comment]
Closed via #235
